mobile_wallet/src/lib.rs

```diff
@@ -349,6 +325,98 @@ fn sign_message_aux(input: &str) -> anyhow::Result<String> {
     ))?)
 }
 
+#[derive(common::SerdeDeserialize)]
+struct MicroCCDAmount {
+    #[serde(rename = "microGtuAmount")]
+    amount: u64,
+}
+
+#[derive(common::SerdeDeserialize)]
+struct AccountAddressJSON {
+    address: String,
+}
+
+#[derive(common::SerdeDeserialize)]
+#[serde(untagged)]
+enum JSONPayload {
+    UpdateContract {
+        amount:       MicroCCDAmount,
+        #[serde(rename = "contractAddress")]
+        address:      ContractAddress,
+        #[serde(rename = "receiveName")]
+        receive_name: String,
+        parameter:    String, // Hex encoded bytes
+        #[serde(rename = "maxContractExecutionEnergy")]
+        energy:       u64,
+    },
+    SimpleTransfer {
+        amount: MicroCCDAmount,
+        to:     AccountAddressJSON,
+    },
+}
+
+/// Context to create an unsigned transaction.
+#[derive(common::SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+struct TransactionContext {
+    pub from:             AccountAddressJSON,
+    pub expiry:           TransactionTime,
+    pub nonce:            Nonce,
+    #[serde(rename = "type")]
+    pub transaction_type: i32,
```

Comment:
What is this? There is a type called `TransactionType` in concordium-base, which has a reasonable JSON instance. So please use that.

Reply:
It is also unclear to me why this is needed since the `JSONPayload` is already an enum.

Reply:
Yeah, I guess we don't need it at all, we use it JS to know how the payload's structure shouldbe.